### ISSUE TITLE
feat(material/datepicker): Apply aria-current="date" to the current date's cell

### ISSUE THREAD

**Issue by ByzantineFailure** [angular/components#23713]
#### Feature Description

Apply `aria-current="date"` to the current date's cell (if present) within mat-datepicker

#### Use Case

[`aria-current` is used to indicate a visually-styled "current" item within a widget](https://www.w3.org/TR/wai-aria-1.2/#aria-current).  As of now, the current date is indicated within `mat-datepicker` via a visual styling but this information is not exposed programmatically to screen readers.

Adding this attribute will bring visual and screen reader information to parity.
